src/compositions/useRoute.ts

```diff
@@ -1,13 +1,15 @@
-import { ExtractRouterRouteParamTypes, RouterRoute } from '@/types'
-import { Identity } from '@/types/utilities'
+import { useRouter } from '@/compositions/useRouter'
+import { UseRouteInvalidError } from '@/errors'
+import { RegisteredRouteMap, ResolvedRoute } from '@/types'
 
-type Route<T extends RouterRoute> = {
-  name: string,
-  params: ExtractRouterRouteParamTypes<T>,
-  query: unknown,
-  hash: string,
-}
+export function useRoute<TRouteName extends keyof RegisteredRouteMap>(routeName: TRouteName): ResolvedRoute<TRouteName>
+export function useRoute(routeName: string): ResolvedRoute {
+  const router = useRouter()
+  const route = router.find(routeName)
 
-export function useRoute<T extends RouterRoute>(): Identity<Route<T>> {
-  throw 'not implemented'
+  if (!route || route.name !== routeName) {
+    throw new UseRouteInvalidError(routeName, router.route.name)
+  }
```

Comment:
Few thoughts here

1. I don't think we actually need to look up the route for `routeName`. We can check against `route.matches`. 
2. We don't need to throw if the `routeName` is a parent of the current route. The user could use `useRoute('settings')` even if the current route is `settings.user`
3. Should this be reactive if the route changes? Like if the user asks for `settings` and when the component mounts that's true, but then the route changes and its no longer `settings`?